src/library/Numbers.nim

```diff
@@ -392,37 +393,48 @@ proc defineSymbols*() =
             #=======================================================
             push(newInteger(int(ceil(asFloat(x)))))
 
-    # TODO(Numbers/clamp) use a Range instead of min/max, as parameters
-    #  given how elegantly Range values are used, it would be better to use e.g a `range` 
-    #  Range type parameter for this one instead of a `max` and `min` parameter, so that we
-    #  we could do something along the lines of `clamp 2 1..3` (would return 2) or 
-    #  `clamp 4 1..3` (would return 3) or `clamp 0 1..3` (would return 1); same as before.
-    #  Also, should we reverse the order of the params, and have the range as the first one?
-    #  labels: library, enhancement
+
     builtin "clamp",
         alias       = unaliased, 
         op          = opNop,
         rule        = PrefixPrecedence,
         description = "force value within given range",
         args        = {
-            "number" : {Integer,Floating},
-            "min"    : {Integer,Floating},
-            "max"    : {Integer,Floating}
+            "number" : {Integer, Floating},
+            "range"  : {Range}
         },
         attrs       = NoAttrs,
-        returns     = {Integer,Floating},
+        returns     = {Integer, Floating},
         example     = """
-            clamp 2 1 3             ; 2
-            clamp 0 1 3             ; 1
-            clamp 4 1 3             ; 3
+            clamp 2 1..3                ; 2
+            clamp 0 1..3                ; 1
+            clamp 4 1..3                ; 3
+            clamp 4 3..1                ; 3
+            clamp 5 range.step: 2 1 5   ; 4
```

Comment:
```suggestion
            clamp 5 range.step: 2 1 5   ; 4
```

This is exactly the part that made me test the code myself... but I think it's just a typo. It should returns (*and* returns) 5, no? 😄

Reply:
Yes, it was a mistyping, my bad. The range should be in this case: `range.step: 2 0 5`, since it generates the following array: `[0 2 4]`.